config/unicorn.rb

```diff
@@ -1,3 +1,17 @@
+# Make sure elasticsearch is configured correctly
+require 'rake'
+begin
+  Rake.application.init
+  Rake.application.load_rakefile
+  Rake.application['search:validate_index'].invoke
+rescue => e
+  # Magic exit code expected by forum-supervisor.sh for when
+  # rake search:validate_index fails
+  puts "ElasticSearch configuration validation failed. "\
+       "\"rake search:validate_index\" failed with the following message: #{e.message}"
+  exit(101)
+end
```

Comment:
@maxrothman: One plan for unit test is to move your code into a unicorn_helpers.rb module contained in an exit_on_invalid_index() function.
Pro:
- simple to test this new method and make sure it does what it is supposed to.
Con:
- It introduces another file with 1 call.  (So?)
- It won't have a test to ensure that the unicorn.rb file doesn't lose the call to exit_on_invalid_index call.  I'm not sure how to test the unicorn.rb file without doing things I don't want to do.
Thoughts?  I want to make sure only one of us are in here as I (or you) refactor so we don't run into conflicts.